src/LinuxTracing/LinuxTracingUtils.cpp

```diff
@@ -213,4 +222,121 @@ bool SetMaxOpenFilesSoftLimit(uint64_t soft_limit) {
   return true;
 }
 
+// Check for the existence of a file mapping containing the absolute address of the function. We
+// have to consider the possibility that the module may be mapped multiple times, and hence that the
+// function may have multiple absolute addresses.
+//
+// Note: A more naive solution would be to look for a map containing the file offset for the
+// function, hence not involving absolute addresses and modules at all. For misaligned PEs, this can
+// cause false negatives, because a function can be mapped twice, in a file mapping and in an
+// anonymous (executable) mapping, but with the actual absolute address of the function being in the
+// anonymous (executable) mapping.
+//
+// Example: Consider a PE with one section, the .text section, at offset in the file 0x400 and
+// relative virtual address 0x1000.
+// The maps could look like this:
+// 140000000-140001000 r--p 00000000 103:07 6946834    /path/to/pe.exe
+// 140001000-140004000 r-xp 00000000 00:00 0
+// The first map corresponds to the headers, however, it also covers all functions with offsets in
+// the file from 0x400 to 0x1000 (i.e., with RVAs from 0x1000 to 0x1c00). But those functions are
+// mapped again in the anonymous map, and that's where they actually have their absolute address,
+// i.e., where they actually get executed.
+static bool FunctionIsInFileMapping(const std::vector<LinuxMemoryMapping>& file_path_maps,
+                                    const std::vector<ModuleInfo>& file_path_modules,
+                                    const InstrumentedFunction& function) {
+  for (const ModuleInfo& module : file_path_modules) {
+    ORBIT_CHECK(module.file_path() == function.file_path());
+    const uint64_t function_absolute_address =
+        orbit_module_utils::SymbolVirtualAddressToAbsoluteAddress(
+            function.function_virtual_address(), module.address_start(), module.load_bias(),
+            module.executable_segment_offset());
+    for (const LinuxMemoryMapping& map : file_path_maps) {
+      ORBIT_CHECK(map.pathname() == function.file_path());
+      if (map.start_address() <= function_absolute_address &&
+          function_absolute_address < map.end_address()) {
```

Comment:
Maybe here I should also check that the function's offset in the file is contained in the file offset range of the map?

Reply:
Intuitively, I would have said that the existing check is sufficient, but it would probably not hurt to also check the offset. Why do you think that check would be needed?

Reply:
I don't think it's needed, conceptually this should be enough. The reason for using the absolute addresses is to avoid false negatives as explained in the comment above, but ultimately it's still about file mappings and file offsets, so I considered this for completeness, but I'm not sure.